NitroxModel/Logger/Log.cs

```diff
@@ -1,148 +1,114 @@
+            logger.Warn(message);
         }
 
-        public static void Error(string fmt, params object[] arg)
+        public static void Error(string message)
         {
-            log.Error(Format(fmt, arg));
+            logger.Error(message);
         }
 
-        public static void Error(string msg, Exception ex)
+        public static void Fatal(string message)
         {
-            log.Error(msg, ex);
+            logger.Fatal(message);
         }
 
-        public static void Warn(string msg)
+        // Sensitive
+        [Conditional("DEBUG")]
+        public static void DebugSensitive(string message, params object[] args)
         {
-            log.Warn(msg);
+            LogFullSensitiveInfo(message, args);
+            Debug(message);
         }
 
-        public static void Warn(string fmt, params object[] arg)
+        public static void InfoSensitive(string message, params object[] args)
         {
-            log.Warn(Format(fmt, arg));
+            LogFullSensitiveInfo(message, args);
+            Info(message);
         }
 
-        public static void Info(string msg)
+        public static void ErrorSensitive(string message, params object[] args)
         {
-            log.Info(msg);
-        }
+            LogFullSensitiveInfo(message, args);
+            Error(message);
+        } 
 
-        public static void Info(string fmt, params object[] arg)
+        public static void Exception(string message, Exception ex)
         {
-            log.Info(Format(fmt, arg));
+            logger.Error(ex, message);
         }
 
-        public static void Info(object o)
+        // In game messages
+        public static void ShowInGameMessage(string message, bool containsPersonalInfo = false)
         {
-            string msg = o == null ? "null" : o.ToString();
-            Info(msg);
+            if (inGameLogger == null)
+            {
+                logger.Warn("InGameLogger has not been registered");
+                return;
+            }
+
+            if (!inGameMessagesEnabled)
+            {
+                logger.Warn("InGameMessages have not been enabled");
+            }
+            inGameLogger.Log(message);
+            
+            // Only log locally if it contains no personal information
+            if (!containsPersonalInfo)
+            {
+                logger.Debug(message);
+            }
+            
         }
 
-        // Only for debug prints. Should not be displayed to general user.
-        // Should we print the calling method for this for more debug context?
-        public static void Debug(string fmt, params object[] arg)
+        public static void RegisterInGameLogger(InGameLogger gameLogger)
         {
-            log.Debug(Format(fmt, arg));
+            logger.Info("Registered InGameLogger");
+            inGameLogger = gameLogger;
         }
 
-        public static void Debug(object o)
+        public static void SetInGameMessagesEnabled(bool enabled)
```

Comment:
Why change this method when it's only ever enabled?

Reply:
So back to your earlier comment about not changing things that already exist.. I noticed this method is called to enable but never called to disable.. I thought about removing this but then in the future someone might want to disable it. I just didnt want to lost functionality that already existed

Reply:
In this case it was designed to only be enbled and never disabled. It limits things that can go wrong (accidental disable and don't know why game messages stop working).

It's best to not support features if we don't need them at all to reduce technical debt.

Reply:
Okay that's fair i shall revert this and remove the boolean